### PR TITLE
added ability to specify a hook which will be called on DB update

### PR DESCRIPTION
**Pull request by AndorCS**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -28,7 +28,13 @@ exports.open = function(filepath, opts, cb) {
     }
 
     if (opts && !!opts.watchForUpdates) {
+      if (opts.watchForUpdatesHook && typeof opts.watchForUpdatesHook != 'function') {
+        throw new Error('opts.watchForUpdatesHook should be a function');
+      }
       fs.watch(filepath, function() {
+        if (opts.watchForUpdatesHook) {
+          opts.watchForUpdatesHook();
```
Comment (runk) [inline index.js]: Should it happen AFTER the reload? Like around L40?

File: test/index.js [hunk 2]
```diff
@@ -141,5 +157,19 @@ describe('index', function() {
...
+
+    it('should successfully handle database updates and call specified hook', function() {
+      var hook = sinon.spy(function() {});
+      var opts = {
+        watchForUpdates: true,
+        watchForUpdatesHook: hook,
+      };
+      var lookup = maxmind.openSync(dbPath, opts);
+      assert(lookup.get('2001:230::'));
+      console.log(hook.notCalled);
```
Comment (runk) [inline test/index.js]: Replace `console.log` with `assert` please

File: test/index.js [hunk 3]
```diff
@@ -141,5 +157,19 @@ describe('index', function() {
       watchHandler();
       assert(fs.readFileSync.calledTwice);
     });
+
+    it('should successfully handle database updates and call specified hook', function() {
+      var hook = sinon.spy(function() {});
```
Comment (runk) [inline test/index.js]: `sinon.spy();` should be sufficient

File: index.js (hunk 1 above)
Comment (AndorCS) [reply index.js]: Oh, sure you are right!

File: test/index.js (hunk 2 above)
Comment (AndorCS) [reply test/index.js]: Oops, artifact of debugging tests :)

**Comment by AndorCS** [comment]
PR updated.

**Comment by runk** [comment]
https://github.com/runk/node-maxmind/releases/tag/v2.6.0 is out

**Pull request merged**